apps/api/src/app/events/usecases/process-subscriber/process-subscriber.usecase.ts

```diff
@@ -1,161 +1,56 @@
-          organizationId: command.organizationId,
-        },
-        command.actor
-      );
-    }
-
-    const notification = await this.createNotification(command, template._id, subscriber);
+  constructor(private createSubscriberUsecase: CreateSubscriber, private subscriberRepository: SubscriberRepository) {}
 
-    const steps: NotificationStepEntity[] = await this.filterSteps.execute(
-      DigestFilterStepsCommand.create({
-        subscriberId: subscriber._id,
-        payload: command.payload,
-        steps: template.steps,
-        environmentId: command.environmentId,
-        organizationId: command.organizationId,
-        userId: command.userId,
-        templateId: template._id,
-        notificationId: notification._id,
-      })
-    );
-
-    const jobs: Omit<JobEntity, '_id'>[] = [];
+  public async execute(command: ProcessSubscriberCommand): Promise<SubscriberEntity | undefined> {
+    const { environmentId, organizationId, subscriber } = command;
 
-    for (const step of steps) {
-      if (!step.template) throw new ApiException('Step template was not found');
+    const subscriberEntity = await this.getSubscriber(environmentId, organizationId, subscriber);
 
-      const providerId: string | undefined = await this.getProviderId(command, step.template.type);
-      jobs.push({
-        identifier: command.identifier,
-        payload: command.payload,
-        overrides: command.overrides,
-        step,
-        transactionId: command.transactionId,
-        _notificationId: notification._id,
-        _environmentId: command.environmentId,
-        _organizationId: command.organizationId,
-        _userId: command.userId,
-        _subscriberId: subscriber._id,
-        status: JobStatusEnum.PENDING,
-        _templateId: notification._templateId,
-        digest: step.metadata,
-        type: step.template.type,
-        providerId: providerId,
-        ...(actorSubscriber && { _actorId: actorSubscriber._id }),
-      });
+    if (subscriberEntity === null) {
+      return undefined;
     }
 
-    return jobs;
-  }
-
-  private async getProviderId(command: ProcessSubscriberCommand, stepType: StepTypeEnum): Promise<string | undefined> {
-    const channelType = STEP_TYPE_TO_CHANNEL_TYPE.get(stepType);
-    if (!channelType) return;
-    const integrations = await this.getDecryptedIntegrations.execute(
-      GetDecryptedIntegrationsCommand.create({
-        channelType: channelType,
-        active: true,
-        organizationId: command.organizationId,
-        environmentId: command.environmentId,
-        userId: command.userId,
-      })
-    );
-    const integration = integrations[0];
-
-    return integration?.providerId ?? InAppProviderIdEnum.Novu;
-  }
-  @Cached(CacheKeyPrefixEnum.NOTIFICATION_TEMPLATE)
-  private async getNotificationTemplate({ _id, environmentId }: { _id: string; environmentId: string }) {
-    return await this.notificationTemplateRepository.findById(_id, environmentId);
```

Comment:
I didn't move this code at all and removed it because we are passing the template all the time to the notification creation so not needed to cache its retrieval as we only need to look for it once.
Before it made sense as we were checking in the loop for every subscriber if the template existed.